sdks/python/apache_beam/runners/test/test_dataflow_runner.py

```diff
@@ -25,14 +25,16 @@
 class TestDataflowRunner(DataflowRunner):
```

Comment:
All of this logic feels like it belongs in TestPipeline, not a subclass of DataflowRunner. @markflyhigh